lib/rubygems/commands/push_command.rb

```diff
@@ -98,19 +89,7 @@ def send_gem(name)
       terminate_interaction 1
     end
 
-    gem_data = Gem::Package.new(name)
-
-    unless @host
-      @host = gem_data.spec.metadata['default_gem_server']
-    end
```

Comment:
With the if branch above, host is not nil (I think? or it would take some serious effort to make it thus), and default_gem_server is already handled

Reply:
To make the refactoring even easier to follow, I extracted this particular refactoring and explanation to its own commit at https://github.com/rubygems/rubygems/pull/2725/commits/e3ccff3c5cd19f56dacadc9a82c9c4ccb89383c7.